src/buildSummary.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 The MathWorks, Inc.
+import * as core from "@actions/core";
+import { join } from 'path';
+import { readFileSync } from 'fs';
+
+export interface Task {
+  name: string;
+  description: string;
+  failed: boolean;
+  skipped: boolean;
+  duration: string;
+}
+
+
+export function getBuildSummaryTable(tasks: Task[]): string[][] {
+  const header: string[] = ['MATLAB Build Task', 'Status', 'Description', 'Duration (HH:MM:SS)'];
+  let taskSummaryTableRows: string[][] = [header];
+
+  tasks.forEach((task, index) => {
+    let taskDetails: string[] = [];
+    taskDetails.push(task.name);
+    if (task.failed) {
+      taskDetails.push('🔴 FAILED');
+    } else if (task.skipped) {
+      taskDetails.push('🔵 SKIPPED');
+    } else {
+      taskDetails.push('🟢 SUCCESS');
+    }
+    taskDetails.push(task.description);
+    taskDetails.push(task.duration);
+
+    taskSummaryTableRows.push(taskDetails);
+  });
+
+  return taskSummaryTableRows;
+}
+
+export function writeSummary(taskSummaryTableRows: string[][]) {
+  core.summary
+    .addTable(taskSummaryTableRows)
+    .write();
+}
```

Comment:
If possible, I still think that we should try to catch errors as soon as we can rather than letting them bubble up to the top level and then giving a generic blanket error message. 

For example, I believe that `write` could throw here so we might as well catch it as close to the error as possible since nothing else cares about a return value. This also let's us provide more specific and helpful error messages.

Reply:
I feel the code block above write is more susceptible for error than write itself. Hence I feel penetrating error  through parent function would make more sense.